test/integration/scheduler/plugins/plugins_test.go

```diff
@@ -2600,101 +2615,41 @@ func TestActivatePods(t *testing.T) {
 	}
 }
 
-var _ framework.PreEnqueuePlugin = &SchedulingGatesPluginWithEvents{}
-var _ framework.EnqueueExtensions = &SchedulingGatesPluginWithEvents{}
-var _ framework.PreEnqueuePlugin = &SchedulingGatesPluginWOEvents{}
-var _ framework.EnqueueExtensions = &SchedulingGatesPluginWOEvents{}
-
-const (
-	schedulingGatesPluginWithEvents = "scheduling-gates-with-events"
-	schedulingGatesPluginWOEvents   = "scheduling-gates-without-events"
-)
-
-type SchedulingGatesPluginWithEvents struct {
-	called int
-	schedulinggates.SchedulingGates
-}
-
-func (pl *SchedulingGatesPluginWithEvents) Name() string {
-	return schedulingGatesPluginWithEvents
-}
-
-func (pl *SchedulingGatesPluginWithEvents) PreEnqueue(ctx context.Context, p *v1.Pod) *framework.Status {
-	pl.called++
-	return pl.SchedulingGates.PreEnqueue(ctx, p)
-}
-
-func (pl *SchedulingGatesPluginWithEvents) EventsToRegister() []framework.ClusterEventWithHint {
-	return []framework.ClusterEventWithHint{
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Update}},
-	}
-}
-
-type SchedulingGatesPluginWOEvents struct {
-	called int
-	schedulinggates.SchedulingGates
-}
-
-func (pl *SchedulingGatesPluginWOEvents) Name() string {
-	return schedulingGatesPluginWOEvents
-}
-
-func (pl *SchedulingGatesPluginWOEvents) PreEnqueue(ctx context.Context, p *v1.Pod) *framework.Status {
-	pl.called++
-	return pl.SchedulingGates.PreEnqueue(ctx, p)
-}
-
-func (pl *SchedulingGatesPluginWOEvents) EventsToRegister() []framework.ClusterEventWithHint {
-	return nil
-}
-
-// This test helps to verify registering nil events for schedulingGates plugin works as expected.
-func TestSchedulingGatesPluginEventsToRegister(t *testing.T) {
-	testContext := testutils.InitTestAPIServer(t, "preenqueue-plugin", nil)
-
-	num := func(pl framework.Plugin) int {
-		switch item := pl.(type) {
-		case *SchedulingGatesPluginWithEvents:
-			return item.called
-		case *SchedulingGatesPluginWOEvents:
-			return item.called
-		default:
-			t.Error("unsupported plugin")
-		}
-		return 0
-	}
+// This test helps to verify registering nil events for plugin works as expected.
+func TestRegisteringNilEvents(t *testing.T) {
```

Comment:
what do you mean by `nil`? Do you mean no hints?

Reply:
For some plugins, e.g. SchedulingGate, it only cares about the update events of pod itself, so it should not watch for pod update events at all. However before, once plugin doesn't register events, we'll register the default events for them by force. So now we allow a plugin to register an empty/nil event (return nil directly), then when building queueingHitMap, once we detect that the `Len(events)` is zero, we'll skip registering to indicate that this plugin isn't interest with any events. 

Add to the description for explanation.